tycho-common/src/simulation/protocol_sim.rs

```diff
@@ -106,23 +106,27 @@ pub trait ProtocolSim: fmt::Debug + Send + Sync + 'static {
     /// Returns the fee of the protocol as ratio
     ///
     /// E.g. if the fee is 1%, the value returned would be 0.01.
+    ///
+    /// # Panics
+    ///
+    /// Panic for protocols with asymmetric fees (e.g. Rocketpool, Uniswap V4 with hooks),
+    /// where a single fee value cannot represent the protocol's fee structure.
     fn fee(&self) -> f64;
```

Comment:
This leaves no option for querying fee without panics. Imho this is good enough reason to introduce a breaking change and make the return value an Option/Result. I don't like being forced to use `std::panic::catch_unwind` to handle the panic.
```suggestion
    fn fee(&self) -> Option<f64>;
```